python/src/ops.cpp

```diff
@@ -3587,6 +3587,52 @@ void init_ops(nb::module_& m) {
             array: The output containing elements selected from
             ``x`` and ``y``.
       )pbdoc");
+  m.def(
+      "nan_to_num",
+      [](const ScalarOrArray& a,
+         std::variant<std::monostate, int, float>& nan_,
+         std::variant<std::monostate, int, float>& posinf_,
+         std::variant<std::monostate, int, float>& neginf_,
```

Comment:
On second thought these can be `std::optional<float>` and I believe `int` will properly cast to `float`.

Then you can pass them directly to the C++ API and handle the type-based max selection there.